Feature asset inventory changes

lomamech opened the pull request:
- Add property _is_asset, external_reference_number, manufacturer_brand, manufacturer_model_, in table **inventory**
- Use table **inventory_type** instead creat table **equipement_type**
- Complete integration test
- And constaint in form of inventory creation, allow reference number,
  manufacturer brand and manufacturer model only of Asset

closes #6348

lomamech committed:
test(Inventory and Inventory type)
- Complete integration test
- And constaint in form of inventory creation, allow reference number,
  manufacturer brand and manufacturer model only of Asset

lomamech committed:
Resolve conflict

lomamech committed:
Clean and sanitaze code

jmcameron commented on client/src/modules/inventory/configuration/types/types.html:
```diff
@@ -40,16 +40,19 @@
-          <td>{{ type.text }}</td>
           <td>
-            <a style="cursor:pointer; text-decoration:none;"
+            <span ng-if="type.is_predefined"><strong>{{ type.text }}</strong></span>
```
I see that you took care of this later

lomamech commented:
> When I edit an inventory item (via its action menu 'Edit' link) and I save the modifications, the Inventory registry loses my columns definitions. The columns definitions should be cached.

To keep the columns visible, click on the menu then on the Column button
![image](https://user-images.githubusercontent.com/8035588/153876549-c8364515-4318-4c00-8d27-31f4f0f568a6.png)

Then you have to choose the columns to display
![image](https://user-images.githubusercontent.com/8035588/153876774-3a912984-270f-4903-b401-898e762a345a.png)

And to save the configuration you just have to click on the **Save Grid Configuration** button
![image](https://user-images.githubusercontent.com/8035588/153877182-ed2c4d09-82bd-4943-b417-6e751a756837.png)

lomamech committed:
improvement(Asset Inventory)
- Use reference_number instead external_reference_number
- Change site for column text in table inventory type`

lomamech commented:
@jmcameron i need a new review, it took me longer because I had also anticipated with the issue #6349 **_Asset Management - Stock Lot changes_**

jmcameron commented:
> And to save the configuration you just have to click on the **Save Grid Configuration** button !

Okay.  I tested this and it works fine.  However, I still wonder why we need the "Save Grid Configuration" function instead of caching any changes to the grid immediately.   I think this is a broader "usability" issue and should not slow down this PR.

lomamech commented:
> > And to save the configuration you just have to click on the **Save Grid Configuration** button !
> 
> Okay. I tested this and it works fine. However, I still wonder why we need the "Save Grid Configuration" function instead of caching any changes to the grid immediately. I think this is a broader "usability" issue and should not slow down this PR.

This functionality exists on all the grids of the BHIMA system and each user can save the grid, the size, the visibility and the positioning of the columns

jmcameron reviewed (commented):
I tried doing the migration procedure and ran into several problems:

- In the migration.sql, file, the change of inventory_type.text cannot include the 'AFTER' clause.  That makes sense because it is modifying an existing field, not added a new field.
- The definition of inventory_type.text needs to be updated in schema.sql (to VARCHAR(200) like the migration change).
- In the migration file (line 57), 'external_reference_number' needs to be changed to 'reference_number'

Please make these updates and I'll try the migration procedure again.

- The definitions in admin.sql of zRecalculateCostCenterAggregates() and zRecomputeLotBarcodes() need to be preceded by DROP PROCEDURE IF EXISTS <name>

lomamech committed:
improvement(Migrate file)
- Update size of inventory_type.text in schema.sql
- Fix error in migrate file us reference_number instead
  external_reference_number
- Add DROP PROCEDURE IF EXISTS for some Procedure

jmcameron commented on server/models/admin.sql:
```diff
@@ -265,6 +265,7 @@ END$$
 
  Removes all data from the cost_center_aggregate table and rebuilds it.
 */
+DROP PROCEDURE IF EXISTS zRecalculateCostCenterAggregates$$
```
Sorry, my mistake.  This addition was not necessary.  This statement is already in the SQL files somewhere.  When you create the migration file, you can find it.

jmcameron reviewed (commented):
I tested migrating an production dataset with this PR and it worked okay.   I also tested displaying and editing asset-related data in the inventory registry and that worked fine.

Before releasing I suggest eliminating the extra DROP PROCEDURE (see the SQL code comments) and adding a DROP PROCEDURE fro zRecomputeLotBarcodes() in admin.sql.

I think this PR is ready to go, but I encourage getting an review by @jniles or @mbayopanda before merging.

jniles commented on client/src/modules/inventory/list/list.html:
```diff
@@ -88,6 +88,8 @@
       ui-grid-save-state
       ui-grid-exporter
       ui-grid-auto-resize
+      ui-grid-resize-columns
```
:+1:

jniles reviewed (approved):
This looks good to me.  Thanks @lomamech 

bors r+
